### DISCUSSION TITLE
Any Wayland support in the near future?

### DISCUSSION THREAD

**Discussion by ghost** [rustdesk/rustdesk#4471] [Feature Request]
It's been a couple weeks that I started using RustDesk and ditched the shit show called TeamViewer, but since my main machine is Debian 11 (Wayland) I have a serious problem with the support of Wayland.

I only login with Wayland and not X11, and since I have a ton of programs open due to work, it's extremely frustrating that I have to close everything and relogin again with X11 to be able to type on the keyboard while assisting someone.

I understand the problems with Wayland security that makes it difficult to implement a complete solution (heck even VirtualBox had some problems with it not a long time ago), but since I started using RustDesk frequently and I use Wayland I would love to know if there's any news about the issue.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/releases/tag/nightly

**Comment by ghost** [comment]
> https://github.com/rustdesk/rustdesk/releases/tag/nightly

Thank you for the info!

**Comment by boospy** [comment]
I tried to connect to a Wayland device here with the current version 1.2 (self hosted). Unfortunately without success so far. 2 Notebook, and one VM for Test. Desktop is KDE Neon. Should it be working already? Thanks :1st_place_medal:

**Reply by rustdesk** [reply]
I just tested KDE Neon, it does not work on it. I just noticed KDE Neon use x11 by default, so I guess its wayland may be too new or not stable enough, we have no plan to adapt KDE Neon wayland fo the time being. So far we only tested on mainstream stable Ubuntu (22.04), Arch, Fedora.

**Reply by boospy** [reply]
Interesting, so KDE does something different... ;( Thanks for testing.
